Objects/longobject.c

```diff
@@ -737,6 +737,16 @@ bits_in_digit(digit d)
     return d_bits;
 }
 
+static int
+popcount_digit(digit d)
+{
+    /* 32bit SWAR popcount. */
+    d -= (d >> 1) & 0x55555555;
+    d = (d & 0x33333333) + ((d >> 2) & 0x33333333);
+    d = (d + (d >> 4)) & 0x0f0f0f0f;
+    return (d * 0x01010101) >> 24;
```

Comment:
The `digit` type could possibly be only two bytes on some platforms. In that case you'd end up potentially overflowing and giving undefined behaviour here. I think you need to either cast the incoming digit to a type big enough for the calculations, or have a dedicated separate version of `popcount_digit` for the case of 15-bit digits.

Reply:
Actually, on second thoughts I think we're okay here, but I'd still recommend converting to either `unsigned long`, or perhaps better `uint32`, at the top of this function.